### ISSUE TITLE
(Chrome) bearded spice isn't working from now

### ISSUE THREAD

**Issue by haiglex** [beardedspice/beardedspice#826]
Hey there, 

thank you for this very useful app on Mac!!! I love it.
I love it so much that now, it doesn't work and it makes me crazy cuz it is so useful. 
I have uninstall/reinstall but nothing works on browser.
perfectly works on iTunes or VLC but not chrome (68.0.3440.106 ((Build officiel) (64 bits)) or safari (11.1.2 (13605.3.8)).

Can you help?

**Comment by agileapricot** [comment]
With the newest update of Chrome/Safari you have to View/Developer/Allow JavaScript from Apple Events to make BeardedSpice work, just as described in the issue 818

**Comment by haiglex** [comment]
Where can I find the step by step details about this issue 818?

**Comment by haiglex** [comment]
Ok I figured it out for safari but not yet on chrome.
The "allow Javascript from Apple Events" is checked already but not working

**Comment by mliq** [comment]
@haiglex worked for me on Chrome, did you try closing and restarting BeardedSpice?

**Comment by haiglex** [comment]
@mliq 
I have just clean the chrome's cache, use Maclean. then reboot the Mac. 
At first, using Soundcloud, it didn't worked, then I opened youtube and like with magie, it turned on beardedspice... don't know why.
Thanks god I can use again this awesome app!!

Thanks guys for the support.

**Comment by Stelminator** [comment]
https://github.com/beardedspice/beardedspice/issues/818 / https://github.com/beardedspice/beardedspice/wiki/Won't-work-issue-after-Chrome-Update-(68.0.3440.75-and-later) worked for me, after restarting both BeardedSpice and Chrome.

**Comment by nzcoops** [comment]
Confirming.

Mac, Chrome, and Google Play Musc

Chrome Version 68.0.3440.106 (Official Build) (64-bit)
High Sierra 10.13.6

Followed above steps, View/Developer/Allow JavaScript from Apple Events, quit BS, reopen BS. 
Working again.

Thanks for making/maintaining one of the most useful mac apps ever.

**Comment by rkrv** [comment]
Confirming this as well.

**Comment by kevindashgit** [comment]
Also confirming @agileapricot  @Stelminator comment/link fixes things for me as well! phew! I was so worried! 😓 ❤️

**Comment by SturmB** [comment]
**OS:** macOS High Sierra 10.13.5
**BeardedSpice:** 2.2.3
**Chrome:** Version 69.0.3497.100 (Official Build) (64-bit)
**View/Developer/Allow JavaScript from Apple Events:** Checked.
**Status:** "No applicable tabs open".

I ran into this problem around the same time as everyone else and, thankfully, the fix suggested in this thread worked perfectly. However, since a few days ago, it has reverted to not working again despite the fact that the "Allow JavaScript from Apple Events" setting is still enabled.

Any ideas?

**Comment by victordevsquad** [comment]
Thanks @agileapricot! Your solution worked like a charm! ;)

**Comment by Timmmm** [comment]
@victordevsquad What OS are you on, and what version of Chrome? It isn't working on OSX with Chrome 69.

**Comment by Timmmm** [comment]
Running in the terminal I get this output:

```
2018-10-25 16:09:43.197 BeardedSpice[98029:5446829] App com.google.Chrome is running <NSRunningApplication: 0x7fa5de802230 (com.google.Chrome - 96593)>
2018-10-25 16:09:44.215 BeardedSpice[98029:5446828] .sdef warning for argument 'FileType' of command 'save' in suite 'Standard Suite': 'saveable file format' is not a valid type name.
```

**Comment by Timmmm** [comment]
Possible solution [here](https://stackoverflow.com/a/5793169/265521).

**Comment by Timmmm** [comment]
Yep that `nonce-...` means that you can't execute any javascript without tagging it with the nonce value, and I expect it means you can't execute any javascript URLs at all (which is how BeardedSpice works). I don't think this will be easy to fix without making a Chrome extension.

I found a couple of extensions for this: [StreamKeys](https://chrome.google.com/webstore/detail/streamkeys/ekpipjofdicppbepocohdlgenahaneen?hl=en), [Web Page Media Keys](https://chrome.google.com/webstore/detail/web-page-media-keys/gndhohcpfmjgggkgkfafpchcnefnhhmc). The latter appears to be open source but has several forks and I couldn't work out which one is the latest.

Also both ask for access to all your websites, which seems unnecessary... Especially as Chrome extension have a habit of being sold to sketchy people.

**Comment by shaypal5** [comment]
Still not working on Chrome on OSX...

**Comment by fharper** [comment]
Not working for me either. It fixed the issue for some YouTube video (not all!?) by [enabling AppleEvent JavaScript](https://github.com/beardedspice/beardedspice/wiki/Won't-work-issue-after-Chrome-Update-(68.0.3440.75-and-later)), but not all.

**Comment by gstorck** [comment]
> @mliq
> I have just clean the chrome's cache, use Maclean. then reboot the Mac.
> At first, using Soundcloud, it didn't worked, then I opened youtube and like with magie, it turned on beardedspice... don't know why.
> Thanks god I can use again this awesome app!!
> 
> Thanks guys for the support.

This worked for me. Thanks. Had to restart BeardedSpice a couple times though! 
(Mac Sierra 10.12.6, Chrome Version 78.0.3904.70)

But I just found out my friend here has a Mac Mojave and the Mac media keys works as default to control music on any browser tab. **Without** any third party app ¬¬

**Comment by miguelvr** [comment]
it works if you enable javascript from apple events in (view → developer → allow javascript from apple events ) and then restart BeardedSpice

**Comment by gevgeny** [comment]
In my case Chrome just quits after I "allow javascript from apple events" and start the beardedspice. Chrome v90.
